src/Deposit.js

```diff
@@ -1019,8 +1044,8 @@ export default class Deposit {
             {},
             true
           )
-        }
-      )
+        })
+        .catch(this.notifyError)
```

Comment:
Why not take an `onError` or similar parameter to `autoSubmit`? It could be optional if someone wanted to handle errors directly on the promise, and we could also emit the event.

In general, a hybrid mechanism where we move to promise workflow + events depending on what the consumer wants to use might be a nice transition for a future direction.

Reply:
Great point. Handled in https://github.com/keep-network/tbtc.js/pull/119